### PR TITLE
User can find artist and see their top tracks

### PR DESCRIPTION
**Pull request by pierre-1**
PT: https://www.pivotaltracker.com/story/show/171702381

```
As a user,
In order to find an artist to listen to
I need to be able to search the artist in a form field and see their top tracks
```
## Changes proposed in this pull request:
* Adds search functionality for artist search and returns their top tracks
* Adds Cypress tests for happy and sad path of user can search for artists
* Formats code
## What I have learned working on this feature:
* <div> id's need to be unique for Cypress tests to find them  
* Cypress testing: calling on an array in a json fixture file taken from the output of a curl command run in API

### CODE REVIEW

File: src/components/UserCanSearchSong.jsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import React, { Component } from "react";
+import axios from "axios";
+
+class UserCanSearchSong extends Component {
+  state = {
```
Comment (oliverochman) [inline src/components/UserCanSearchSong.jsx]: same comment as before

File: cypress/fixtures/artist_data.json [hunk 2]
```diff
@@ -0,0 +1,576 @@
+{
```
[the remark this answers is not in the document]
Comment (pierre-1) [reply cypress/fixtures/artist_data.json]: We have removed the file

File: cypress/fixtures/artist_top_songs_data.json [hunk 3]
```diff
@@ -0,0 +1,764 @@
+{
```
[the remark this answers is not in the document]
Comment (pierre-1) [reply cypress/fixtures/artist_top_songs_data.json]: Json response array has been modified with sanitized artist data

File: cypress/integration/userCanFindArtist.feature.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
+describe("user can search for artist", () => {
+  describe("happy path", () => {
+    beforeEach(() => {
+      cy.server();
+      cy.route({
+        method: "GET",
+        url: "http://localhost:3000/api/v1/tracks**",
```
[the remark this answers is not in the document]
Comment (pierre-1) [reply cypress/integration/userCanFindArtist.feature.js]: Artist controller has been called on

File: cypress/integration/userCanFindArtist.feature.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    });
+
+    it("retrieves artist from API", () => {
+      cy.get("input#search-field").type("U2");
+      cy.get("button#search").click();
+      cy.get("#artistName").should("contain", "U2");
+    });
+  });
+
+  describe("SAD PATH user can search for song", () => {
```
[the remark this answers is not in the document]
Comment (pierre-1) [reply cypress/integration/userCanFindArtist.feature.js]: Update completed

File: cypress/integration/userCanFindArtist.feature.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    beforeEach(() => {
+      cy.server();
+    });
+    it("User invalid search", () => {
+      cy.route({
+        method: "GET",
+        url: "http://localhost:3000/api/v1/tracks**",
+        status: "400",
+        response: {
+          song_not_found: [
```
[the remark this answers is not in the document]
Comment (pierre-1) [reply cypress/integration/userCanFindArtist.feature.js]: Modified to error_message

File: src/App.jsx [hunk 7]
```diff
@@ -1,60 +1,27 @@
 import React, { Component } from "react";
-import axios from "axios";
+import UserCanSearchSong from "./components/UserCanSearchSong";
+import UserCanSearchArtist from "./components/UserCanSearchArtist";
 
 class App extends Component {
   state = {
```
[the remark this answers is not in the document]
Comment (pierre-1) [reply src/App.jsx]: State has been updated

File: src/components/UserCanSearchArtist.jsx [hunk 8]
```diff
@@ -0,0 +1,63 @@
+import React, { Component } from "react";
+import axios from "axios";
+
+class UserCanSearchArtist extends Component {
+  state = {
```
[the remark this answers is not in the document]
Comment (pierre-1) [reply src/components/UserCanSearchArtist.jsx]: State has been updated

File: src/components/UserCanSearchSong.jsx (hunk 1 above)
Comment (pierre-1) [reply src/components/UserCanSearchSong.jsx]: State has been updated

File: cypress/integration/userCanFindArtist.feature.js [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+      cy.get("#artistName").should("contain", "U2");
+    });
+  });
+
+  describe("SAD PATH user can search for song", () => {
+    beforeEach(() => {
+      cy.server();
+    });
+    it("User invalid search", () => {
+      cy.route({
```
[the remark this answers is not in the document]
Comment (pierre-1) [reply cypress/integration/userCanFindArtist.feature.js]: Change implemented by extracting beforeEach for sad path.

**Pull request merged**